.github/scripts/lint-ecosystem.js

```diff
@@ -16,26 +16,36 @@ module.exports = async function ({ core }) {
   const moduleNameRegex = /^\- \[\`(.+)\`\]/
   let hasOutOfOrderItem = false
   let lineNumber = 0
-  let inCommmunitySection = false
+  let inCommunitySection = false
   let modules = []
+  let hasImproperFormat = false
 
   for await (const line of rl) {
     lineNumber += 1
     if (line.startsWith('#### [Community]')) {
-      inCommmunitySection = true
+      inCommunitySection = true
     }
     if (line.startsWith('#### [Community Tools]')) {
-      inCommmunitySection = false
+      inCommunitySection = false
     }
-    if (inCommmunitySection === false) {
+    if (inCommunitySection === false) {
       continue
     }
 
-    if (line.startsWith('- [`') !== true) {
+    if (line.startsWith('- [') !== true) {
```

Comment:
Okay, this makes sense. We were just skipping the line if it started like `- [f`.

Reply:
Indeed, cases like ` - [foo]`, ``- [foo`]`` were undetectable.
In regard to extending the script functionality, do you wish that I make another commit with it? (as I stated, `Core` and `Community Tools` were being validated, but I've reverted this part in my last commit)

Reply:
Yes, that would be a new pull request with an improvement. I originally left those out because they are generally only added by the core Fastify team. We don't usually have out-of-order issues with those. But it doesn't hurt to add the validation.

Reply:
Alright then, should I then open another PR with this state? [`70027c57c1`](https://github.com/nooreldeensalah/fastify/blob/70027c57c12c64d0846fa415a7ce2e6c007472e2/.github/scripts/lint-ecosystem.js) (i.e., before the revert commit). Or are there any other changes you would like to point out to?

Reply:
I'd rather just review the feature in isolation instead of trying to pick it out of other changes.

Reply:
I've opened another PR at https://github.com/fastify/fastify/pull/4258